tools/analyse.sh

```
@@ -0,0 +1,264 @@
+mapfile -t PROJECT_FILES < "${DB_FILES}"
+
+if [ "${#PROJECT_FILES[@]}" -eq 0 ]; then
+  echo -e "${YELLOW}[WARN]${NC} No source files found in compilation database"
+else
+  echo -e "${BLUE}[INFO]${NC} Analysing ${#PROJECT_FILES[@]} source files from compilation database..."
+  for src_file in "${PROJECT_FILES[@]}"; do
+    clang-tidy \
+      --config-file="${APP_DIR}/.clang-tidy" \
+      --header-filter="${APP_DIR}/src/.*" \
+      -p "${PROJECT_DB}" \
+      ${ARM_GCC_INCLUDES} \
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)